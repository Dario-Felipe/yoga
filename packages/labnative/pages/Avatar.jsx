import React from 'react';
import styled from 'styled-components';
import { Avatar } from '@gympass/yoga';
import { Youtube, UserFilled } from '@gympass/yoga-icons';
import { DocTitle } from '../components';
import avatar from '../assets/images/avatarIcons/avatar.png';
import classIcon from '../assets/images/avatarIcons/class.png';

const ScrollView = styled.ScrollView`
  margin-bottom: 50px;
  padding: 10px;
  width: 100%;
`;

const AvatarWrapper = styled.View`
  width: 100%;
  padding: 20px;
  margin-bottom: 10px;
  align-items: center;
  justify-content: space-between;
  height: 250px;
  flex-direction: column;
`;

const AvatarPage = () => (
  <ScrollView>
    <AvatarWrapper>
      <DocTitle>Default Avatar without image prop</DocTitle>
      <Avatar />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Default Avatar</DocTitle>
      <Avatar src={avatar} />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Default Avatar with placeholder prop</DocTitle>
      <Avatar placeholder={Youtube} />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Default Avatar with width and height props</DocTitle>
      <Avatar src={avatar} width={78} height={78} />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Circle Avatar without image prop</DocTitle>
      <Avatar.Circle />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Circle Avatar </DocTitle>
      <Avatar.Circle src={classIcon} />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Circle Avatar with placeholder prop</DocTitle>
      <Avatar.Circle placeholder={UserFilled} />
    </AvatarWrapper>
    <AvatarWrapper>
      <DocTitle>Circle Avatar with width and height props</DocTitle>
      <Avatar.Circle src={classIcon} width={78} height={78} />
    </AvatarWrapper>
  </ScrollView>
);

export default AvatarPage;
