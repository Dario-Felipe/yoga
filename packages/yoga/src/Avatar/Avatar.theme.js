const Avatar = ({ spacing, colors }) => ({
  backgroundColor: colors.medium,
  color: colors.white,
  icon: {
    size: spacing.medium,
  },
});

export default Avatar;
