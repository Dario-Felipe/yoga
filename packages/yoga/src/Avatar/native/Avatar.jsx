import React from 'react';
import { BuildingFilled } from '@gympass/yoga-icons';
import { string, func, number } from 'prop-types';
import Box from '../../Box';
import Image from './Image';
import Placeholder from './Placeholder';

const Avatar = ({
  src,
  fill,
  placeholder,
  width = 48,
  height = 48,
  borderRadius = 'regular',
}) => {
  const icon = placeholder || BuildingFilled;
  return (
    <Box
      bgColor="elements.selectionAndIcons"
      display="flex"
      width={width}
      height={height}
      alignItems="center"
      justifyContent="center"
      elevation="small"
      borderRadius={borderRadius}
    >
      {src ? <Image source={src} /> : <Placeholder fill={fill} icon={icon} />}
    </Box>
  );
};

Avatar.propTypes = {
  src: string,
  placeholder: func,
  fill: string,
  width: number,
  height: number,
  borderRadius: string,
};

Avatar.defaultProps = {
  src: undefined,
  placeholder: undefined,
  fill: undefined,
  width: undefined,
  height: undefined,
  borderRadius: undefined,
};

Avatar.displayName = 'Avatar';

export default Avatar;
