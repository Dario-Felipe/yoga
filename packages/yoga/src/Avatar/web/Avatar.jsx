import React from 'react';
import { BuildingFilled } from '@gympass/yoga-icons';
import { string, func, checkPropTypes } from 'prop-types';

import Box from '../../Box';
import Icon from '../../Icon';

/**
 * The Avatar component is used to display the image.
 * It has two shapes: default and circle
 */
const Avatar = ({ src, alt, fill, stroke, icon, ...otherProps }) => (
  <Box
    bgColor="elements.selectionAndIcons"
    display="flex"
    alignItems="center"
    justifyContent="center"
    width={48}
    height={48}
    borderRadius="small"
    elevation="small"
    overflow="hidden"
    {...otherProps}
  >
    {src ? (
      <img src={src} alt={alt} />
    ) : (
      <Icon as={icon} width="50%" height="50%" fill={fill} stroke={stroke} />
    )}
  </Box>
);

Avatar.propTypes = {
  src: string,
  /**  Required if src is true */
  alt: (props, propName, componentName) => {
    const { src } = props;

    if (src) {
      checkPropTypes(
        { [propName]: string.isRequired },
        props,
        'prop',
        componentName,
      );
    }

    return null;
  },
  icon: (props, propName, componentName) => {
    const { src } = props;

    if (src) {
      return new Error(
        `Invalid prop ${propName} supplied to ${componentName}.`,
      );
    }

    checkPropTypes({ [propName]: func }, props, 'prop', componentName);

    return null;
  },
  fill: string,
  stroke: string,
  ...Box.propTypes,
};

Avatar.defaultProps = {
  src: undefined,
  alt: undefined,
  icon: BuildingFilled,
  fill: 'white',
  stroke: undefined,
  ...Box.defaultProps,
};
Avatar.displayName = 'Avatar';

export default Avatar;
