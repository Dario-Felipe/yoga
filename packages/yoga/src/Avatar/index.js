import Avatar from './web/Avatar';
import AvatarCircle from './web/AvatarCircle';

Avatar.Circle = AvatarCircle;

export default Avatar;
