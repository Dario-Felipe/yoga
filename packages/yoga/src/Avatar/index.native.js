import Avatar from './native/Avatar';
import AvatarCircle from './native/AvatarCircle';

Avatar.Circle = AvatarCircle;

export default Avatar;
