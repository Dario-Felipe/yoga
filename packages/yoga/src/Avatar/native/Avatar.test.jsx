import React from 'react';
import { render } from '@testing-library/react-native';

import { UserFilled } from '@gympass/yoga-icons';
import { ThemeProvider } from '../..';
import Avatar from './Avatar';

describe('<Avatar />', () => {
  it('should match snapshot in default avatar', () => {
    const { container } = render(
      <ThemeProvider>
        <Avatar width={78} height={78} fill="black" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot in default avatar with placeholder prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Avatar placeholder={UserFilled} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot in circle avatar', () => {
    const { container } = render(
      <ThemeProvider>
        <Avatar.Circle />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot in circle avatar with placeholder prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Avatar.Circle placeholder={UserFilled} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
