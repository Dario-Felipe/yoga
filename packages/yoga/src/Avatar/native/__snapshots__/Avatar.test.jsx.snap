// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should match snapshot in avatar with children prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  height={48}
  justifyContent="center"
  overflow="hidden"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      },
    ]
  }
  width={48}
>
  <Text
    fontSize="medium"
    inverted={false}
    light={false}
    style={
      [
        {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
      ]
    }
  >
    AL
  </Text>
  <View
    borderColor="rgba(35, 27, 34, 0.1)"
    borderRadius="small"
    borderWidth={0.5}
    height="100%"
    style={
      [
        {
          "borderColor": "rgba(35, 27, 34, 0.1)",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;

exports[`<Avatar /> should match snapshot in circle avatar 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="circle"
  display="flex"
  height={48}
  justifyContent="center"
  overflow="hidden"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 9999,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    aria-hidden={true}
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "height": "50%",
            "width": "50%",
          },
        ],
        {
          "opacity": 1,
        },
        {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={{}}
      matrix={
        [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
  <View
    borderColor="rgba(35, 27, 34, 0.1)"
    borderRadius="circle"
    borderWidth={0.5}
    height="100%"
    style={
      [
        {
          "borderColor": "rgba(35, 27, 34, 0.1)",
          "borderRadius": 9999,
          "borderWidth": 0.5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;

exports[`<Avatar /> should match snapshot in circle avatar with placeholder prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="circle"
  display="flex"
  height={48}
  justifyContent="center"
  overflow="hidden"
  placeholder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 9999,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    aria-hidden={true}
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "height": "50%",
            "width": "50%",
          },
        ],
        {
          "opacity": 1,
        },
        {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={{}}
      matrix={
        [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
  <View
    borderColor="rgba(35, 27, 34, 0.1)"
    borderRadius="circle"
    borderWidth={0.5}
    height="100%"
    style={
      [
        {
          "borderColor": "rgba(35, 27, 34, 0.1)",
          "borderRadius": 9999,
          "borderWidth": 0.5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;

exports[`<Avatar /> should match snapshot in default avatar 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  height={78}
  justifyContent="center"
  overflow="hidden"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 78,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 78,
      },
    ]
  }
  width={78}
>
  <RNSVGSvgView
    aria-hidden={true}
    bbHeight="50%"
    bbWidth="50%"
    fill="black"
    focusable={false}
    height="50%"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "height": "50%",
            "width": "50%",
          },
        ],
        {
          "opacity": 1,
        },
        {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        [
          0,
          4278190080,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={{}}
      matrix={
        [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
  <View
    borderColor="rgba(35, 27, 34, 0.1)"
    borderRadius="small"
    borderWidth={0.5}
    height="100%"
    style={
      [
        {
          "borderColor": "rgba(35, 27, 34, 0.1)",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;

exports[`<Avatar /> should match snapshot in default avatar with placeholder prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  height={48}
  justifyContent="center"
  overflow="hidden"
  placeholder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    aria-hidden={true}
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "height": "50%",
            "width": "50%",
          },
        ],
        {
          "opacity": 1,
        },
        {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={{}}
      matrix={
        [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
  <View
    borderColor="rgba(35, 27, 34, 0.1)"
    borderRadius="small"
    borderWidth={0.5}
    height="100%"
    style={
      [
        {
          "borderColor": "rgba(35, 27, 34, 0.1)",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;
